rollup.config.js

```diff
@@ -0,0 +1,19 @@
+import copy from "rollup-plugin-copy";
+
+export default {
+    input: "src/index.js",
+    output: [
+        {
+            file: "dist/esm/index.js",
+            format: "esm",
+            banner: '// @ts-self-types="./index.d.ts"'
+        }
+    ],
+    plugins: [
+        copy({
+            targets: [
+                { src: "src/types.ts", dest: "dist/esm" }
+            ]
+        })
+    ]
```

Comment:
Is using Rollup + plugins a settled choice? I've had an easier time with [tsup](https://tsup.egoist.dev) as a more TypeScript-native builder.

Reply:
It's what we've been using in all of the type-checked repos so far. Given that it's working for us, I don't think this PR is the place to ponder alternatives. However, it is a bit clunky, so an exploration of another option would be welcome as a separate PR.